app/jobs/regular/translate_translatable.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Jobs
+  class TranslateTranslatable < ::Jobs::Base
+    def execute(args)
+      return unless SiteSetting.translator_enabled
+      return if SiteSetting.automatic_translation_target_languages.blank?
+
+      translatable = args[:type].constantize.find_by(id: args[:translatable_id])
+      return if translatable.blank?
+
+      target_locales = SiteSetting.automatic_translation_target_languages.split("|")
+      target_locales.each do |target_locale|
+        # 1. no special retry, job will be automatically retried with backoff
+        # 2. translate function will handle cases where translation is not needed or not possible
+        "DiscourseTranslator::#{SiteSetting.translator}".constantize.translate(
+          translatable,
+          target_locale.to_sym,
+        )
+      end
+
+      topic_id = translatable.is_a?(Post) ? translatable.topic.id : translatable.id
+      post_id = translatable.is_a?(Post) ? translatable.id : 1
```

Comment:
```suggestion
    topic_id, post_id = translatable.is_a?(Post) ? [translatable.topic_id, translatable.topic.id] : [translatable.id, 1]
 ```
 
 Minor but after that we will have only a single check for translatable.is_a?(Post)
 Also translatable.topic_id instead of translatable.topic.id is better because topic does not have to be loaded